### DISCUSSION TITLE
Struggling to get Whisper installed on an M1 Macbook Pro

### DISCUSSION THREAD

**Discussion by jonathanbossenger** [openai/whisper#403] [General]
I follow the installation steps for an m1 chip, but when I try to run whisper I get the error:

```
zsh: command not found: whisper
```

These are the steps I followed to install whisper:

Ran [the commands from the issue](https://github.com/openai/whisper/discussions/160) to install setuptools_rust and upgrade pip
```
sudo pip3 install setuptools-rust
sudo pip3 install --upgrade pip
```

Ran the following command to install whisper

```
python3 -m pip install git+https://github.com/openai/whisper.git
```

Installed ffmpeg via homebrew

```
brew install ffmpeg
```

I did some digging in the discussions and I [did find this thread](https://github.com/openai/whisper/discussions/162). So I checked my `/Library/Python/3.9/` directory, but it does not have a `bin` directory. I also checked `/Library/Frameworks`, but it does not have a `Python.framework` directory. 

I'm thinking whisper is being installed, and it just needs to be added to the system `$PATH`, but I just have no idea where it's installed, or how to check where it's installed.

**Comment by jonathanbossenger** [comment]
I managed to get this working by using [pyenv](https://github.com/pyenv/pyenv)

## Install pyenv
```
brew update
brew install pyenv
```

## install python version 3.9.14 (latest) and set it to the global version
```
pyenv install 3.9.14
pyenv global 3.9.14
```

## test things are installed correctly
```
python --version
```   
Should output `Python 3.9.14`

```
pip --version
```
Should output something like `pip 22.0.4 from /Users/{homeuser}/.pyenv/versions/3.9.14/lib/python3.9/site-packages/pip (python 3.9)`

Then running `pip install git+https://github.com/openai/whisper.git` installed whisper correctly.

**Reply by jsjoeio** [reply]
Thank you for documenting this! I had to edit my `~/.zshrc` file and add this:
```shell
export PYENV_ROOT="$HOME/.pyenv"
export PATH="$PYENV_ROOT/bin:$PATH"

eval "$(pyenv init --path)"
eval "$(pyenv init -)"
```

Then `python --version` and everything else worked. 

Source: https://stackoverflow.com/a/71646176/3015595

**Comment by voidfel** [comment]
I am curious, why didn't you just install it through brew? Works fine on Macbook Pro M1 Pro.

```shell
brew update
brew install openai-whisper
```

**Reply by jonathanbossenger** [reply]
Probably because at the time [it wasn't available as a brew formula](https://github.com/Homebrew/homebrew-core/commit/67302b7bd4e80f56bd8f13340de43a5c7af3be46), and the only [setup instructions in the readme](https://github.com/openai/whisper#setup) used the manual installation method.

**Reply by alejaramos** [reply]
it worked perfect on my pro M1

**Comment by supernacho** [comment]
After install `brew install openai-whisper ` the executable is created in  /opt/homebrew/bin/whisper

To be recognized by the system:
`export PATH="/opt/homebrew/bin:$PATH"`
